torch/_subclasses/fake_tensor.py

```diff
@@ -589,13 +582,35 @@ def __init__(
         # normalize cuda device.
         if device.type == "cuda" and device.index is None:
             device = torch.device(f"cuda:{torch.cuda.current_device()}")
-        self.fake_device = device
-        self.fake_mode = fake_mode
-        self.constant = constant
+        self.fake_device = device  # type: ignore[attr-defined]
+        self.fake_mode = fake_mode  # type: ignore[attr-defined]
+        self.constant = constant  # type: ignore[attr-defined]
         if FakeTensorConfig.debug:
             import traceback
 
-            self._debug_trace = traceback.extract_stack()
+            self._debug_trace = traceback.extract_stack()  # type: ignore[attr-defined]
+        return self
+
+    # In some circumstances, a conventional torch.Tensor constructor
```

Comment:
this is a bit confusing but I don't know of a better way.

Reply:
Note that if `__new__` returns an unrelated object, the `__init__` call does get skipped, but not if it's a subclass: https://gist.github.com/ezyang/8a570c4d86cae9d5d67e9dc84acd2c56